third_party/nvidia/backend/driver.py

```diff
@@ -518,6 +519,7 @@ def __call__(self, gridX, gridY, gridZ, stream, function, *args):
             grid_size = gridX * gridY * gridZ
             alloc_size = grid_size * self.global_scratch_size
             global_scratch = _allocation._allocator(alloc_size, self.global_scratch_align, stream)
+            set_alloc_state(global_scratch, grid_size, self.global_scratch_size, self.global_scratch_align, stream)
```

Comment:
Yeah, it might be better to just add a new `profile_buffer` argument to the launch function

Reply:
This way `global_scratch` can be used as it was

Reply:
And we can profile kernels with both `global_scratch` and `profile_buffer`.

Reply:
My initial thought was actually something like

```
def alloc_fn():
    size = size + profile_buffer_size
    ret = torch.empty(size, device="cuda", dtype=torch.int8)
    libproton.set_profile_buffer(original pointer of ret + size)
```

But still it's not as clean as using an independent buffer